src/jit/lclvars.cpp

```diff
@@ -255,13 +301,25 @@ void Compiler::lvaInitTypeRef()
         varDsc->lvPinned  = ((corInfoType & CORINFO_TYPE_MOD_PINNED) != 0);
         varDsc->lvOnFrame = true; // The final home for this local variable might be our local stack frame
 
+        CORINFO_CLASS_HANDLE clsHnd = info.compCompHnd->getArgClass(&info.compMethodInfo->locals, localsSig);
+        stackSize += roundUp(compGetTypeSize(strip(corInfoType), clsHnd), TARGET_POINTER_SIZE);
+
```

Comment:
You hoisted this out of the ` if (strip(corInfoType) == CORINFO_TYPE_CLASS)` condition. Is it legal (or desirable) to call `getArgClass` in all cases?